### PR TITLE
Feat/UI ux enhancement framegear

### PR DESCRIPTION
**Pull request by adarshswaminath**
# Enhance Home Component with Layout, Loading State, and Placeholder Improvements

## What changed? Why?
This PR includes the following changes:

1. **Improved Responsiveness and Layout**:
   - Updated the grid layout in the `Home` component to ensure better responsiveness: now a single column on smaller screens and two columns on larger screens.
   - Added padding (`p-2`) to the main container for better spacing and improved layout on all screen sizes.

2. **Loading State in FrameInput Component**:
   - Added a loading state to the `fetchFrame` API call in `FrameInput`. The button now reflects "Loading..." when the API call is in progress, enhancing user experience.

3. **Placeholder for Missing Images**:
   - Added placeholders to handle missing images, ensuring the UI remains clean and informative even when images are unavailable.

## Notes to reviewers
- Please focus on checking the responsiveness of the layout across different screen sizes.
- Verify that the loading state in the `FrameInput` component works as expected.
- Check if the placeholders for missing images are working correctly.

## How has it been tested?
- Manual testing was performed to ensure the layout is responsive on both small and large screens.
- The loading state was tested by simulating a slow API call, confirming the button updates appropriately.
- Missing images were tested by intentionally breaking image links, verifying the placeholders display properly.

## Screenshots:

image
image
image

### CODE REVIEW

File: framegear/components/Frame/Frame.tsx [hunk 1]
```diff
@@ -40,8 +40,8 @@ function ValidFrame({ metadata }: { metadata: FrameMetadataWithImageObject }) {
     <div>
       {/* eslint-disable-next-line @next/next/no-img-element */}
       <img
-        className={`w-full rounded-t-xl ${imageAspectRatioClassname} object-cover`}
-        src={image.src}
+        className={`w-full rounded-t-xl ${imageAspectRatioClassname} ${image.src ? "object-cover" : "object-contain"}`}
+        src={image.src ? image.src : "https://demofree.sirv.com/nope-not-here.jpg?w=750"}
```
Comment (Zizzamia) [inline framegear/components/Frame/Frame.tsx:44]: Sorry we can not have links here on random websites.

File: framegear/components/FrameInput/FrameInput.tsx [hunk 2]
```diff
@@ -28,12 +31,12 @@ export function FrameInput() {
         />
       </label>
       <button
-        className="h-[40px] self-end rounded-full bg-white text-black"
+        className={`h-[40px] self-end rounded-full ${isLoading ? "bg-neutral-800 text-white" : "bg-white text-black"}`}
```
Comment (Zizzamia) [inline framegear/components/FrameInput/FrameInput.tsx:34]: I believe you can use `clx` on Framegear as well.

**Comment by Zizzamia** [comment]
I am going to have @brendan-defi review this as well, as he is leading our effort to keep polish Framegear.

File: framegear/components/Frame/Frame.tsx (hunk 1 above)
Comment (adarshswaminath) [reply framegear/components/Frame/Frame.tsx:44]: would it be acceptable if I add the images directly to the public folder of the project and reference them from there?

File: framegear/components/FrameInput/FrameInput.tsx (hunk 2 above)
Comment (adarshswaminath) [reply framegear/components/FrameInput/FrameInput.tsx:34]: Okay, I will implement the `clx`

File: framegear/components/Frame/Frame.tsx (hunk 1 above)
Comment (Zizzamia) [reply framegear/components/Frame/Frame.tsx:44]: yup, but only if it's a cool image. If it's not, is probably best have something else as default.

Comment (adarshswaminath) [reply framegear/components/Frame/Frame.tsx:44]: image
 is it ok ?

Comment (Zizzamia) [reply framegear/components/Frame/Frame.tsx]: I think is best we avoid add those kind of images so far. As we want our designer to take a look at it.

Comment (adarshswaminath) [reply framegear/components/Frame/Frame.tsx]: Ok, I'll update the image based on the designer's review.

File: framegear/components/Frame/Frame.tsx [hunk 3]
```diff
@@ -76,10 +85,6 @@ function ValidFrame({ metadata }: { metadata: FrameMetadataWithImageObject }) {
 }
 
 function ErrorFrame() {
-  // TODO: implement -- decide how to handle
```
Comment (brendan-defi) [inline framegear/components/Frame/Frame.tsx:79]: Definitely acknowledge that this isn't elegant, but let's not remove. We will work on cleaning up these comments.

File: framegear/components/Frame/Frame.tsx [hunk 4]
```diff
@@ -116,14 +121,12 @@ function FrameButton({
 
   const handleClick = useCallback(async () => {
     if (button?.action === 'post' || button?.action === 'post_redirect') {
-      // TODO: collect user options (follow, like, etc.) and include
```
Comment (brendan-defi) [inline framegear/components/Frame/Frame.tsx:119]: as above, let's keep this comment

File: framegear/components/Frame/Frame.tsx [hunk 5]
```diff
@@ -116,14 +121,12 @@ function FrameButton({
...
       const confirmAction = async () => {
         const result = await postFrame(
           {
             buttonIndex: index,
             url: (button as any).postUrl!,
             state: JSON.stringify(state),
-            // TODO: make these user-input-driven
```
Comment (brendan-defi) [inline framegear/components/Frame/Frame.tsx:126]: same here

File: framegear/components/Frame/Frame.tsx [hunk 6]
```diff
@@ -136,7 +139,6 @@ function FrameButton({
           },
           mockFrameOptions
         );
-        // TODO: handle when result is not defined
```
Comment (brendan-defi) [inline framegear/components/Frame/Frame.tsx:139]: same here

File: framegear/components/Frame/Frame.tsx [hunk 7]
```diff
@@ -162,7 +164,6 @@ You can test this action on the official Warpcast validator: https://warpcast.co
 (must deploy frame to a publicly accessible URL)`
       );
     }
-    // TODO: implement other actions (mint, etc.)
```
Comment (brendan-defi) [inline framegear/components/Frame/Frame.tsx:165]: same here

**Comment by brendan-defi** [comment]
> I am going to have @brendan-defi review this as well, as he is leading our effort to keep polish Framegear.

Thanks @Zizzamia !

@adarshswaminath Thanks for taking the time to work on this. The responsiveness and the loading state on the fetch-button are nice, good work. :-)

Left a few comments. Most importantly, I don't think we should add an placeholder image like this though. This generic 404 isn't really the right message (the issue isn't necessarily a 404, it's that the image for that frame isn't valid, which can happen for many reasons). With these changes I think you'll be g2g.

Thanks!

**Comment by brendan-defi** [comment]
> @brendan-defi Thankyou for the review :) I don’t know why I removed those comments earlier :( I have now added them back. Instead of the placeholder image, I have added a 'No Image Found' message, which is straightforward for the user to understand that there is no image to display. 

Thanks @adarshswaminath, nice work :-)

**Review by brendan-defi — APPROVED** [review]

**Comment by adarshswaminath** [comment]
Thank you! Actually, these comments from @brendan-defi  and @Zizzamia  are really exciting for a newbie like me. Getting positive feedback from such a great team feels like the next level :-) . Is this PR ready to merge?

**Review by Zizzamia — APPROVED** [review]

**Pull request merged**